core/views.py

```diff
@@ -99,6 +119,35 @@ def goals(request):
         "yp": yp,
         "current_goals": current_goals,
         "complete_goals": complete_goals,
+        "user": user,
+    }
+    return render(request, template_name="core/yp/goals.html", context=context)
+
+
+@login_required
+def yp_goals(request, young_person_id):
+    user = request.user
+    yp = YoungPerson.objects.get(pk=young_person_id)
+    current_goals = yp.goals.filter(completed__isnull=True, archived__isnull=True)
+    complete_goals = yp.goals.filter(completed__isnull=False, archived__isnull=True)
+    action_form = NewActionForm()
+    if request.method == "POST":
+        action_form = NewActionForm(request.POST)
+        if action_form.is_valid():
+            action_form.save(request)
+            messages.success(request, "Action saved.")
+            return redirect("goals")
+        else:
+            messages.error(request, "Action not saved. Invalid information.")
+            return redirect("goals")
+    else:
+        pass
```

Comment:
Pass is a "do nothing" statement so isn't needed here really. I suggest removing the else statement here entirely.